### DISCUSSION TITLE
Is upgrading socket.io required while upgrading socket.io-client

### DISCUSSION THREAD

**Discussion by Abhishek325** [socketio/socket.io#3928] [Q&A] [answered]
Hi folks, I upgraded my `socket.io-client` (from 2.3.0 to 4.1.1)  to address few security issues. My server is still running `socket.io` 2.3.0 and I don't see anything broken with my app. Is the client supposed to be independent of socket version on server?

**Comment by Rc85** [comment]
From my experience, yes, it is required. If you have different versions, they will not connect.

**Comment by darrachequesne** [comment, marked as answer]
I guess it's still working because you are using `transports: ["websocket"]`, in that specific case `socket.io@2` is indeed compatible with `socket.io-client@4`.

That being said, the bump from v2 to v3 was backward incompatible at the protocol level, so a client in v3 (with HTTP long-polling enabled) is not compatible with a server in v2. More information about it [here](https://socket.io/docs/v4/migrating-from-2-x-to-3-0/).

You can find the compatibility table here: https://socket.io/docs/v4/client-installation/#Version-compatibility
